CMakeLists.txt

```diff
@@ -74,6 +78,20 @@ target_compile_definitions(mbed-core
         ${MBED_CONFIG_DEFINITIONS}
 )
 
+# Set compile definitions for the toolchain supported
```

Comment:
Shall we add a note here, these TOOLCHAIN_xxx are for backward compatibility, not recommended to be used in new files. The new code should use proper toolchain checks we use:

```
Gcc Arm: `__GNUC__`
ARMClang: `__clang__` 
IAR: `__ICCARM__`
```

Note for future fix, we can start using toolchain CMake variable (will refactor most of the code), as we do not need this MBED_TOOLCHAIN. One of the things we tried to fix in the older tools as well but never all cleaned up.

Reply:
I can add it but is this the best place to document that?

Reply:
We might not be that verbose, but stating `this is for backward compatibility and these should not be used in new files` should be here. We should check docs if this is mentioned anywhere.

Reply:
https://github.com/ARMmbed/mbed-os/compare/721795010195a507564ef5cb36300c6a7d6d54d5..37c18ff3d066782ee8ca106538b30f2c2d3c4da5